hazelcast/src/test/java/com/hazelcast/jet/core/ExecutionLifecycleTest.java

```diff
@@ -236,6 +297,49 @@ public void when_pmsCloseThrows_then_jobSucceeds() {
         assertsWhenOneJob();
     }
 
+    //ProcessorSupplier throws NonSerializable exception from init()
+    @Test
+    public void when_psInitThrowsNonSerializable_then_jobFails() {
```

Comment:
When I do the proposed changes in the modified asserts I get

```
Execution on a member failed: com.hazelcast.jet.JetException: com.hazelcast.nio.serialization.HazelcastSerializationException: Failed to serialize 'com.hazelcast.spi.impl.operationservice.impl.responses.ErrorResponse'
...
```
and it doesn't contain the stacktrace of the non-serializable exception. We should handle such exception in the code and convert it to string. 

I am not sure if we need to add a new check here 
https://github.com/hazelcast/hazelcast/blob/641cf926d49b2b80ed072371dd9325ca8cb2e99f/hazelcast/src/main/java/com/hazelcast/jet/impl/operation/AsyncOperation.java#L92

or maybe here 
https://github.com/hazelcast/hazelcast/blob/641cf926d49b2b80ed072371dd9325ca8cb2e99f/hazelcast/src/main/java/com/hazelcast/jet/impl/operation/AsyncOperation.java#L102
it should have been value, not `ex` that's converted to string, I don't think converting `ex` there makes much sense
cc @TomaszGaweda

Reply:
After the changes it does not fail anymore on the local environment. I will push one more time and see the results

Reply:
+1, `peel` in that `catch` block  from the second snippet is not peeling the JetException - so it doesn't see underlying HSE. This should be fixed, I've created https://github.com/hazelcast/hazelcast/pull/22889 for the fix